ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+temp_multipliers = [0.82, 0.88, 0.92, 0.95, 0.98, 1, 0.98, 0.91, 0.74,
+                    0.55, 0.38, 0.23, 0.13, 0.05, 0, 0, 0.06, 0.14, 0.24,
+                    0.39, 0.5, 0.59, 0.68, 0.75]
+
+
+class Ddy(object):
+    """Import data from a local .ddy file.
+
+    properties:
+        location: A Ladybug location object
+        design_days: A list of the design days in the ddy file.
+        file_path: Optional file path into which ddy file can be written.
+    """
+    # TODO: set the default file path to use the ladybug default folder.
+    def __init__(self, location, design_days=[], file_path=None):
+        """Initalize the class."""
+        self.location = location
+        self.design_days = design_days
+        self.file_path = file_path or 'C:\\ladybug\\unnamed.ddy'
+
+    @classmethod
+    def from_ddy_file(cls, file_path):
+        """Initalize from a ddy file object from an existing ddy file.
+
+        args:
+            file_path: A full string to the .ddy file.
+        """
+        # defaults
+        location = None
+        design_days = []
+
+        # check that the file is there
+        if not os.path.isfile(file_path):
+            raise ValueError(
+                'Cannot find an ddy file at {}'.format(file_path))
+
+        # check the python version and open the file
+        try:
+            iron_python = True if platform.python_implementation() == 'IronPython' \
+                else False
+        except Exception:
+            iron_python = True
+
+        if iron_python:
+            ddywin = codecs.open(file_path, 'r')
+        else:
+            ddywin = codecs.open(file_path, 'r', encoding='utf-8', errors='ignore')
+
+        try:
+            ddytxt = ddywin.read()
+            location_format = re.compile(
+                r"(Site:Location,(.|\n)*?((;\s*!)|(;\s*\n)|(;\n)))")
+            design_day_format = re.compile(
+                r"(SizingPeriod:DesignDay,(.|\n)*?((;\s*!)|(;\s*\n)|(;\n)))")
+            location_matches = location_format.findall(ddytxt)
+            des_day_matches = design_day_format.findall(ddytxt)
+        except Exception as e:
+            import traceback
+            raise Exception('{}\n{}'.format(e, traceback.format_exc()))
+        else:
+            # build design day and location objects
+            location = Location.from_location(location_matches[0][0])
+            design_days = [DesignDay.from_ep_string(
+                match[0], location) for match in des_day_matches]
+        finally:
+            ddywin.close()
+
+        return cls(location, design_days, file_path)
+
+    @classmethod
+    def from_design_day(cls, design_day):
+        """Initalize from a ddy file object from a ladybug design day object.
+        """
+        return cls(design_day.location, [design_day])
+
+    def save(self, folder=None, file_name=None):
+        """Save ddy object as a .ddy file."""
+        # check file_path
+        if not folder:
+            folder = self.folder
```

Comment:
If you're keeping self.folder for the purpose of saving the file then I think you can just remove it and make folder as a required input. The same about file_name.

Reply:
I'm not sure if I follow the reasoning for why we should remove it.
Is there a reason why we shouldn't build in the convenience of saving a ddy file that has been edited with the API in the same folder as a ddy file where it originated from?

Reply:
This was addressed in our discussion